build-support/bin/generate_github_workflows.py

```diff
@@ -763,6 +763,9 @@ def build_wheels_job(platform: Platform, python_versions: list[str]) -> Jobs:
 
 
 def build_wheels_jobs() -> Jobs:
+    # N.B.: When altering the number of total wheels built (currently 10), please edit the expected
```

Comment:
I cowboyed a fixes for 2.14.x and 2.15.x having incorrect wheel counts after cherry-picks removing the macOS 11 x86_64 wheel builing shard:
+ https://github.com/pantsbuild/pants/commit/bf1880dc68d397a4d3b279a54759a23456d03e5b
+ https://github.com/pantsbuild/pants/commit/be45d8143afcbe923f56478f80c3d63428829e94

This is intended to make that harder to miss.